koku/api/currency/utils.py

```diff
@@ -0,0 +1,41 @@
+#
+# Copyright 2021 Red Hat Inc.
+# SPDX-License-Identifier: Apache-2.0
+#
+import logging
+
+from api.currency.models import ExchangeRateDictionary
+
+LOG = logging.getLogger(__name__)
+
+
+def build_exchange_dictionary(rates, index=0, exchange_rates={}):
+    """Build the exchange rates dictionary"""
+    currency_dict = {}
+    base_currency_list = list(rates.keys())
+    base_currency = base_currency_list[index]
+    for code in rates:
+        if base_currency == "USD":
+            currency_dict[code] = rates[code]
+        else:
+            if code == "USD":
+                currency_dict[code] = float(1 / rates[base_currency])
```

Comment:
float is imprecise. I'd suggest using decimal.Decimal
The math could be different in subsequent runs.